gqlalchemy/loaders.py

```diff
@@ -0,0 +1,386 @@
+# Copyright (c) 2016-2022 Memgraph Ltd. [https://memgraph.com]
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from . import Memgraph
+from .query_builder import QueryBuilder, Unwind
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from gqlalchemy.models import (
+    MemgraphIndex, 
+    MemgraphTrigger, 
+    TriggerEventObject, 
+    TriggerEventType, 
+    TriggerExecutionPhase
+)
+from pyarrow import fs
+from typing import (
+    List, 
+    Dict, 
+    Any, 
+    Optional,
+    Tuple, 
+    Union
+)
+import pyarrow.dataset as ds
+
+
+
+@dataclass(frozen=True)
+class RelationshipMapping:
+    foreign_key: str
+    reference_table: str
+    reference_key: str
+    label: str
+    variables: Optional[Dict[str, str]] = None
+
+
+
+@dataclass(frozen=True)
+class Nx1Mapping:
+    mapping: RelationshipMapping
+    from_entity: bool = False
+
+
+
+@dataclass(frozen=True)
+class MxNMapping:
+    mapping1: RelationshipMapping
+    mapping2: RelationshipMapping
+    label: str
+    from_first: bool = True
+
+
+
+Mapping = Union[List[Nx1Mapping], MxNMapping]
+
+
+
+@dataclass
+class CollectionMapping:
```

Comment:
The naming is kinda misleading since it doesn't have anything to do with python collections :/